pkg/apis/triggers/v1alpha1/event_listener_types.go

```diff
@@ -35,7 +38,72 @@ type TriggerBindingRef struct {
 	APIVersion string `json:"apiversion,omitempty"`
 }
 
-type EventListenerStatus struct{}
+const (
+	// ServiceConditionType is the Service ConditionType set on the EventListener
+	ServiceConditionType apis.ConditionType = "Service"
+	// DeploymentConditionType is the Deployment ConditionType set on the EventListener
+	DeploymentConditionType apis.ConditionType = "Deployment"
+)
+
+var eventListenerCondSet = apis.NewLivingConditionSet(ServiceConditionType, DeploymentConditionType)
+
+// +k8s:deepcopy-gen=true
+type EventListenerStatus struct {
+	duckv1beta1.Status `json:",inline"`
+	// Configuration stores configuration for the EventListener service
+	Configuration EventListenerConfig `json:"configuration"`
+}
+
+type EventListenerConfig struct {
+	// Hostname is the EventListener's internal/local service hostname
+	Hostname string `json:"hostname"`
+}
+
+// GetCondition returns the Condition matching the given type.
+func (el *EventListenerStatus) GetCondition(t apis.ConditionType) *apis.Condition {
+	return eventListenerCondSet.Manage(el).GetCondition(t)
+}
+
+// SetCondition sets the condition, unsetting previous conditions with the same type as necessary.
+// This is a local change and needs to be persisted to the K8s API elsewhere.
+func (el *EventListenerStatus) SetCondition(newCond *apis.Condition) {
+	if newCond != nil {
+		eventListenerCondSet.Manage(el).SetCondition(*newCond)
+	}
+}
+
+// InitializeConditions will set all conditions in eventListenerCondSet to unknown for the EventListener.
+// This does not use the InitializeCondition() provided by the conditionsImpl to avoid setting the happy condition.
+// This is a local change and needs to be persisted to the K8s API elsewhere.
+func (el *EventListenerStatus) InitializeConditions() {
+	for _, condition := range []apis.ConditionType{
+		ServiceConditionType,
+		DeploymentConditionType,
+	} {
+		el.SetCondition(&apis.Condition{
+			Type:   condition,
+			Status: corev1.ConditionUnknown,
+		})
+	}
+}
+
+// SetStatus marks specified Condition true if the passed error is nil, otherwise it is marked false.
+// This is a local change and needs to be persisted to the K8s API elsewhere.
+func (el *EventListenerStatus) SetStatus(t apis.ConditionType, e error) {
+	if e == nil {
+		el.SetCondition(&apis.Condition{
+			Type:    t,
+			Status:  corev1.ConditionTrue,
+			Message: fmt.Sprintf("%s started successfully", t),
+		})
+	} else {
+		el.SetCondition(&apis.Condition{
+			Type:    t,
+			Status:  corev1.ConditionFalse,
+			Message: e.Error(),
+		})
+	}
+}
```

Comment:
its pretty trivial but id write a couple unit tests for these functions - they wouldnt do much right now but it should be super easy to do and they might be useful later!